src/PrestaShopBundle/Service/Command/AbstractCommand.php

```diff
@@ -25,55 +25,49 @@
  */
 namespace PrestaShopBundle\Service\Command;
 
+use AppKernel;
+use Exception;
 use Symfony\Bundle\FrameworkBundle\Console\Application;
 use Symfony\Component\Console\Output\BufferedOutput;
 use Symfony\Component\Console\Input\ArrayInput;
 
 abstract class AbstractCommand
 {
-    private $env;
+    protected $kernel;
     protected $application;
-    protected $commands;
+    protected $commands = [];
 
     /**
      * Constructor.
      *
      * Construct the symfony environment.
      *
-     * @param string $env Environment to set.
+     * @param AppKernel $kernel Symfony Kernel
      */
-    public function __construct($env = null)
+    public function __construct(AppKernel $kernel = null)
     {
-        umask(0000);
```

Comment:
```
Avoid using this function in multithreaded webservers. It is better to change the file permissions with chmod() after creating the file. Using umask() can lead to unexpected behavior of concurrently running scripts and the webserver itself because they all use the same umask.
```
Same for set time limit, settings it's really better to setting up directly with the server.

Reply:
be careful, users are not all sysadmin, the soft has to run without complex tasks server side